hoot-core-test/src/test/cpp/hoot/core/schema/ScoreMatrixTest.cpp

```diff
@@ -73,6 +75,51 @@ class ScoreMatrixTest : public CppUnit::TestFixture
     uut.set(4, 2, .99);
 
     HOOT_STR_EQUALS(1.6, uut.minSumScore());
+
+    uut.resize(5, 3);
+
+    uut(0, 1) = 1;
+    uut(0, 2) = 0.8;
+
+    uut(1, 0) = .8;
+    uut(1, 1) = .98;
+    uut(1, 2) = 0.0;
+
+    uut(2, 0) = 1;
+    uut(2, 1) = .1;
+    uut(2, 2) = .98;
+
+    uut(3, 0) = .5;
+    uut(3, 1) = 1;
+    uut(3, 2) = 1;
+
+    uut(4, 0) = 1;
+    uut(4, 1) = .8;
+    uut(4, 2) = .99;
+
+    HOOT_STR_EQUALS(1.6, uut.minSumScore());
+
+    HOOT_STR_EQUALS("\n"
+      "|   | 0 | 1 | 2 | \n"
+      "| 0 |   0 |    1 |  0.8 | \n"
+      "| 1 | 0.8 | 0.98 |    0 | \n"
+      "| 2 |   1 |  0.1 | 0.98 | \n"
+      "| 3 | 0.5 |    1 |    1 | \n"
+      "| 4 |   1 |  0.8 | 0.99 | \n",
+      uut.toTableString());
+
+    vector<QString> rows( {"<unassigned>", "California", "Hostel", "in", "Hollywood"} );
+    vector<QString> columns( {"<unassigned>", "Hotel", "California"} );
+    uut.setLabels(rows, columns);
+
+    HOOT_STR_EQUALS("\n"
+      "|              | <unassigned> | California | Hotel | \n"
+      "| <unassigned> |            0 |        0.8 |     1 | \n"
+      "|   California |          0.8 |          0 |  0.98 | \n"
+      "|    Hollywood |            1 |       0.99 |   0.8 | \n"
+      "|       Hostel |            1 |       0.98 |   0.1 | \n"
+      "|           in |          0.5 |          1 |     1 | \n",
+      uut.toTableString());
```

Comment:
Here we are testing the setting the column/row headers that make the table look cleaner.